Fetch the versions from Constants.swift and CHANGELOG.md

pepibumur opened the pull request:
### Short description 📝
Getting the list of versions and the latest version through GitHub's API has proven to be a bad idea due to API limits. This PR addresses that by changing the source of truth for that information so that we don't have to go through the API. In particular, we get the latest version from the `Constants.swift` and the list of versions from `CHANGELOG.md`.

Note that although I'm adding the code for getting the versions, it's not being used. We might in the future if we want to validate that a given local version exists remotely. Also, note that I'm keeping `GitHubClient` in case we need it in the future.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

adellibovi commented:
@pepibumur, I may have missed something, but why can't we use `https://github.com/tuist/tuist/releases/latest/download/tuist.zip` ? 🤔

adellibovi commented on Sources/TuistEnvKit/Settings/GitHubVersionController.swift:
```diff
@@ -0,0 +1,140 @@
+            .eraseToAnyPublisher()
+    }
+
+    func latestVersion() -> AnyPublisher<Version, Error> {
```
Maybe we can implement this by calling versions? Or is there any downside?

pepibumur commented:
> are we sure we want to keep the unused code (both the new versions and the old GitHubClient)? I would rather delete it and retrieve it from history if we'll ever need it

That's a good point. I removed the code from the codebase.

> if we want to keep it, are we sure we want to use 2 different approaches to "get all the versions" (parse the changelog) and "get the latest version" (read the Constants.swift)?

I left the versions one and modified the `latestVersion` to be a convenience method that depends on `versions`.
